src/OpenLoco/src/GameCommands/GameCommands.h

```diff
@@ -1613,14 +1613,25 @@ namespace OpenLoco::GameCommands
         }
     };
 
-    // Remove town
-    inline bool do_50(uint8_t townId)
+    struct TownRemovalArgs
     {
-        registers regs;
-        regs.bl = Flags::apply;
-        regs.edi = townId;
-        return doCommand(GameCommand::removeTown, regs) != FAILURE;
-    }
+        static constexpr auto command = GameCommand::removeTown;
+
+        TownRemovalArgs() = default;
+        explicit TownRemovalArgs(const registers& regs)
+            : townId(TownId(regs.edi))
```

Comment:
change this to regs.di as the function looks to only use di so possibly some caller might pass it from di.

Reply:
Hmm on second thoughts it doesn't actually matter the function also does an edi comparison.